### PR TITLE
Ensure all Jelly view have escape-by-default.

### PR DESCRIPTION
**Pull request by Wadeck**
- all those views escape the content theirself or desire to output raw.

@reviewbybees

### CODE REVIEW

File: core/src/main/resources/hudson/model/TextParameterDefinition/config.jelly [hunk 1]
```diff
@@ -22,7 +22,7 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 -->
 
-<?jelly escape-by-default='false'?>
+<?jelly escape-by-default='true'?>
```
Comment (Wadeck) [inline core/src/main/resources/hudson/model/TextParameterDefinition/config.jelly]: For that one, the escaping is done by `textarea.jelly`

**Review by oleg-nenashev — APPROVED** [review]
🐝 . Maybe it makes sense to enhance Injected tests in JTH to support this strict requirement.

**Comment by recampbell** [comment]
@reviewbybees done

File: core/src/main/resources/hudson/model/Computer/_scriptText.jelly [hunk 2]
```diff
@@ -25,6 +25,6 @@ THE SOFTWARE.
 <!--
   Called from doScriptText() to display the execution result.
 -->
-<?jelly escape-by-default='false'?>
+<?jelly escape-by-default='true'?>
 <st:compress xmlns:j="jelly:core" xmlns:st="jelly:stapler">
-<st:contentType value="text/plain;charset=UTF-8" />${output}</st:compress>
\ No newline at end of file
+<st:contentType value="text/plain;charset=UTF-8" /><st:out value="${output}"/></st:compress>
```
Comment (Wadeck) [inline core/src/main/resources/hudson/model/Computer/_scriptText.jelly]: Please review this usage of `st:out`

File: core/src/main/resources/jenkins/model/Jenkins/_scriptText.jelly [hunk 3]
```diff
@@ -25,6 +25,6 @@ THE SOFTWARE.
 <!--
   Called from doScriptText() to display the execution result.
 -->
-<?jelly escape-by-default='false'?>
+<?jelly escape-by-default='true'?>
 <st:compress xmlns:j="jelly:core" xmlns:st="jelly:stapler">
-<st:contentType value="text/plain;charset=UTF-8" />${output}</st:compress>
\ No newline at end of file
+<st:contentType value="text/plain;charset=UTF-8" /><st:out value="${output}"/></st:compress>
```
Comment (Wadeck) [inline core/src/main/resources/jenkins/model/Jenkins/_scriptText.jelly]: and this one

**Review by Wadeck — COMMENTED** [review]
I changed some `st:out` into `j:out` because the content was expected to be tags and the `st:out` would have escaped them, what is not desired.
@oleg-nenashev could you just review the remaining `st:out` (listed below) since I am not sure where they are called from ? (not even if their end points is still used actually)

File: core/src/main/resources/hudson/model/Computer/_scriptText.jelly (hunk 2 above)
Comment (daniel-beck) [reply core/src/main/resources/hudson/model/Computer/_scriptText.jelly]: It sets a content type header, so if the result isn't escaped, then it's fine.

File: core/src/main/resources/hudson/model/DownloadService/footer.jelly [hunk 4]
```diff
@@ -29,7 +29,7 @@ THE SOFTWARE.
 
   This file is pulled into the layout.jelly
 -->
-<?jelly escape-by-default='false'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
-  ${it.generateFragment()}
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <j:out value="${it.generateFragment()}"/>
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/DownloadService/footer.jelly]: I don't really understand what a change like this adds to a file like this one.

- Pro: When someone adds content to the file, people need to remember to escape or perform a change like this.
- Con: Adds complexity, and potential `st:out` / `j:out` confusion.

File: core/src/main/resources/hudson/model/Computer/_scriptText.jelly (hunk 2 above)
Comment (Wadeck) [reply core/src/main/resources/hudson/model/Computer/_scriptText.jelly]: `st:out` escape the `<` and `&`.
`j:out` just render the value without escaping.

File: core/src/main/resources/hudson/model/DownloadService/footer.jelly (hunk 4 above)
Comment (Wadeck) [reply core/src/main/resources/hudson/model/DownloadService/footer.jelly]: It was actually for the Pro. After that commit normally all the Jelly views will be escaped by default. Specific behavior will be explicitly written with the usage of `j:out` for non-escaping content.

I think the confusion is just my fault, when I wrote this PR I was not totally aware of the difference between `st:out` and `j:out`. After your comments, there will be no other `st:out` usage in the PR.

Comment (daniel-beck) [reply core/src/main/resources/hudson/model/DownloadService/footer.jelly]: Well, the confusion remains if someone doesn't remember offhand which is which.

Not a big deal, just wondered.

**Comment by oleg-nenashev** [comment]
On hold for this release

**Comment by oleg-nenashev** [comment]
@daniel-beck are you fine with the merge?

**Comment by daniel-beck** [comment]
@oleg-nenashev No objections.

**Comment by oleg-nenashev** [comment]
@Wadeck if you consider it as a defect or notable change, please follow the PR template

**Comment by Wadeck** [comment]
@Oleg for me it's just a small modification to ensure next analysis on usage of jelly escaping will not find them and require some investigations to see they are not problematic

**Pull request merged**